src/box/sql/build.c

```diff
@@ -303,7 +303,8 @@ sql_shallow_space_copy(struct Parse *parse, struct space *space)
 	}
 	memcpy(ret->index, space->index, size);
 	memcpy(ret->def, space->def, sizeof(struct space_def));
-	ret->def->opts.is_temporary = true;
+	if (!space_is_temporary(ret))
+		ret->def->opts.temporary = SPACE_OPTS_TEMPORARY;
```

Comment:
Unrelated changes like this are the reason why we need to extract the temporary/space_type enum introduction into a preparatory commit.

Reply:
done